### ISSUE TITLE
Removing item with a unique but not required attribute not allowed

### ISSUE THREAD

**Issue by sweepy84** [sensedeep/dynamodb-onetable#317]
It appears that you cannot remove an item if a unique attribute has not been set.

In the example schema below, creating a Project item without a `code` attribute and then trying to remove that item results in the below message.

``` 
const project = Project.create({ name: 'A Project' })
Project.remove({ id: project.id })

> [OneTableArgError: Cannot remove unique field "code" for model "Project", must provide "code" value] 

The only solution is to provide an incorrect code value (an undefined, or empty string also errors out), which seems like odd.

```const ProjectSchema = {
  version: '0.0.1',
  client,
  name: 'ProjectSchema',
  indexes: {
    primary: { hash: 'pk', sort: 'sk' },
    gs1: { hash: 'gs1pk', sort: 'gs1sk' },
  },
  models: {
    Project: {
      pk: { type: String, value: '${_type}#${id}' },
      sk: { type: String, value: '${_type}#' },
      id: { type: String, required: true, generate: 'ulid' },
      code: { type: String, unique: true },
      name: { type: String, required: true },

      gs1pk: { type: String, value: '${_type}#${code}' },
      gs1sk: { type: String, value: '${_type}#' },

    },
  } as const,
}```

**Comment by sweepy84** [comment]
I thought I would fork and tweak the code - but realised it's already been done! [here](https://github.com/sensedeep/dynamodb-onetable/commit/190f28c4e14deef1b2a603de32e70bdbcf7e2f55)

So just need to wait for it to be released! :)

**Comment by sweepy84** [comment]
I might just point out that that implementation still uses a Transaction to write a single item, which will be needlessly expensive.

**Comment by mobsense** [comment]
Thanks for highlighting that about single transaction items. I'll add that to our todo list.

**Comment by mobsense** [comment]
Fixed in 2.3.5
